src/core/timer.c

```diff
@@ -401,12 +401,10 @@ static void timer_enter_waiting(Timer *t, bool time_change) {
 
                         if (t->last_trigger.realtime > 0)
```

Comment:
hmm, i think it would make sense to also use dual_timestamp_is_set() here, to emphasize the symmetry. i.e. 1st choice is the last trigger, second choice the activation, and finally the current time.